apps/web/components/booking/AvailableTimes.tsx

```diff
@@ -132,8 +136,20 @@ const AvailableTimes: FC<AvailableTimesProps> = ({
                 let slotFull, notEnoughSeats;
 
                 if (slot.attendees && seatsPerTimeSlot) slotFull = slot.attendees >= seatsPerTimeSlot;
-                if (slot.attendees && bookingAttendees && seatsPerTimeSlot)
+                if (slot.attendees && bookingAttendees && seatsPerTimeSlot) {
                   notEnoughSeats = slot.attendees + bookingAttendees > seatsPerTimeSlot;
+                }
+
+                const isHalfFull =
+                  slot.attendees && seatsPerTimeSlot && slot.attendees / seatsPerTimeSlot >= 0.5;
+                const isNearlyFull =
+                  slot.attendees && seatsPerTimeSlot && slot.attendees / seatsPerTimeSlot >= 0.83;
```

Comment:
This is duplicate a lot also- Why dont we calculate this higher up - maybe when slots are fetched we can do this calculation instead of doing it on every use case

Reply:
A Nit for sure, but not a blocking one.